packages/auth/src/authClients/supabase.ts

```diff
@@ -32,13 +46,28 @@ export const supabase = (client: Supabase): AuthClientSupabase => {
   return {
     type: 'supabase',
     client,
-    login: ({ email, password }) => client.auth.signIn({ email, password }),
-    logout: () => client.auth.signOut(),
-    signup: ({ email, password }) => client.auth.signUp({ email, password }),
+    login: async ({ email, password, provider }) => {
+      if (email && !password) {
+        // magic link
+        return await client.auth.signIn({ email })
+      }
+      // email and password
+      if (email && password)
+        return await client.auth.signIn({ email, password })
+      // oauth, such as github, gitlab, etc.
+      if (provider) return await client.auth.signIn({ provider })
```

Comment:
```suggestion
      if (provider) {
        return await client.auth.signIn({ provider })
      }
```

Reply:
Sorry, I find these easier to read.

Reply:
Making this requested change for braces around the if.